ols/app/models/config.py

```diff
@@ -277,6 +308,51 @@ def setup_models_config(self, data: dict) -> None:
             model = ModelConfig(m, self.type)
             self.models[m["name"]] = model
 
+    def set_provider_specific_configuration(self, data: dict) -> None:
+        """Set the provider-specific configuration."""
+        # compute how many provider-specific configurations are
+        # found in config file
+        found = 0
+        for x in constants.SUPPORTED_PROVIDER_TYPES:
+            cfg_name = x + "_config"
+            if data.get(cfg_name) is not None:
+                found += 1
+
+        # just none or one provider-specific configuration
+        # should available
+        if found > 1:
+            raise InvalidConfigurationError(
+                "multiple provider-specific configurations found, "
+                f"but just one is expected for provider {self.type}"
+            )
+
+        # If one provider-specific configuration is available
+        # it must match the selected provider type.
+        # It means, that if configuration for selected provider
+        # is not present the configuration must be wrong.
+        if found == 1:
+            match self.type:
+                case constants.PROVIDER_AZURE_OPENAI:
+                    self.azure_config = data.get("azure_openai_config", None)
+                    self.check_provider_config(self.azure_config)
+                case constants.PROVIDER_OPENAI:
+                    self.openai_config = data.get("openai_config", None)
+                    self.check_provider_config(self.openai_config)
+                case constants.PROVIDER_BAM:
+                    self.watsonx_config = data.get("bam_config", None)
+                    self.check_provider_config(self.bam_config)
+                case constants.PROVIDER_WATSONX:
+                    self.watsonx_config = data.get("watsonx_config", None)
+                    self.check_provider_config(self.watsonx_config)
+
```

Comment:
Isn't some raise here missing in case we don't match?

Reply:
we match for sure, because the code to retrieve provider type does this check. But ok, I can add `_` case there

Reply:
updated @onmete